src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs

```diff
@@ -74,6 +75,7 @@ public DebuggedProcess(bool bLaunched, LaunchOptions launchOptions, ISampleEngin
             ExceptionManager = new ExceptionManager(MICommandFactory, _worker, _callback, configStore);
 
             VariablesToDelete = new List<string>();
+            this.ActiveVariables = new List<IVariableInformation>();
```

Comment:
You are accessing this from multiple threads. Probably should have a lock+make this private, or you a collection with locking built in.